### PR TITLE
Tests user inactive

### PR DESCRIPTION
**Pull request by ruanres**
**Feature/Bug description:** It was created tests for:

- UserInactiveController
- SearchInstitution component controller
- UserRequestFormController

**Solution:** ...

**TODO/FIXME:** n/a

### CODE REVIEW

File: frontend/test/specs/user/userRequestFormControllerSpec.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    };
+
+    const request = { 
+        status: 'sent',
+        sender_key: 'some-key'
+    };
+
+    beforeEach(module('app'));
+
+    beforeEach(inject(function($controller, $rootScope, $state,
```
Comment (JuliePessoa) [inline frontend/test/specs/user/userRequestFormControllerSpec.js]: teste $onInit

File: frontend/test/specs/user/userRequestFormControllerSpec.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    }));
+
+    describe('UserRequestFormController functions', function() {
+
+        describe('getMesage()', function () {
+            const formMsg = "Para finalizar o pedido de convite, preencha suas informações institucionais";
+            const requestSentMsg = `Sua solicitação de convite foi enviada e esta em analise pelo administrador
+             de sua instituição na plataforma e-CIS. Voce recebera a confirmação em seu e-mail.`
+
+            it('should return the form message when the was not sent yet', function () {
```
Comment (JuliePessoa) [inline frontend/test/specs/user/userRequestFormControllerSpec.js]: when the request*

File: frontend/test/specs/user/userRequestFormControllerSpec.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                userReqFormCtrl.isRequestSent = false;
+                expect(userReqFormCtrl.getMessage()).toEqual(formMsg);
+            });
+
+            it('should return the request sent message when the was already sent', function () {
```
Comment (JuliePessoa) [inline frontend/test/specs/user/userRequestFormControllerSpec.js]: when the request

File: frontend/test/specs/user/userRequestFormControllerSpec.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                expect(userReqFormCtrl.isRequestSent).toBeFalsy();
+            });
+        });
+
+        describe('onClick()', function () {
+            it('should call the logout function when the request was already sent', function () {
+                spyOn(authService, 'logout');
+                userReqFormCtrl.isRequestSent = true;
+                userReqFormCtrl.onClick();
+                expect(authService.logout).toHaveBeenCalled();
```
Comment (JuliePessoa) [inline frontend/test/specs/user/userRequestFormControllerSpec.js]: also verify that `_verifyAndSendRequest` was not called.

File: frontend/test/specs/user/userRequestFormControllerSpec.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+            });
+
+            it('should logout function when the request was not sent yet', function () {
+                spyOn(userReqFormCtrl, '_verifyAndSendRequest');
+                userReqFormCtrl.isRequestSent = false;
+                userReqFormCtrl.onClick();
+                expect(userReqFormCtrl._verifyAndSendRequest).toHaveBeenCalled();
```
Comment (JuliePessoa) [inline frontend/test/specs/user/userRequestFormControllerSpec.js]: also verify that `authService.logout` was not called.

File: frontend/test/specs/user/searchInstitutionControllerSpec.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+        describe('search()', function(){
+            beforeEach(function () {
+                spyOn(searchInstCtrl, 'onSearch');
+                spyOn(institutionService, 'searchInstitutions').and.returnValue(deferred.promise);
+            });
+            
+            it('Should return the institutions found', function(){
+                const institutions = [institution, otherInstitution];
+                deferred.resolve(institutions);
```
Comment (Luiz-FS) [inline frontend/test/specs/user/searchInstitutionControllerSpec.js]: Using the same promise and solving it twice can be a problem, right?

File: frontend/test/specs/user/userRequestFormControllerSpec.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+                userReqFormCtrl.requests = [
+                    {...request}, 
+                    {...request, sender_key: user.key}
+                ];
+                expect(userReqFormCtrl._wasInstRequested()).toBeTruthy();
+            });
+
+            it('should be false when the user has not requested the institution yet', function () {
+                userReqFormCtrl.requests = [
+                    {...request},
```
Comment (Luiz-FS) [inline frontend/test/specs/user/userRequestFormControllerSpec.js]: Test the case of no request

File: frontend/test/specs/user/searchInstitutionControllerSpec.js (hunk 6 above)
Comment (ruanres) [reply frontend/test/specs/user/searchInstitutionControllerSpec.js]: The beforeEach executes before each test, so this is always reset before a new test.
More information: http://breazeal.com/blog/jasmineBefore.html

File: frontend/user/searchInstitution/searchInstitution.component.js [hunk 8]
```diff
@@ -31,7 +32,8 @@
          * Check if the institution is selected and loaded
          */
         searchInstCtrl.isInstSelected = function (institution){
-            return searchInstCtrl.selectedInst.key === institution.id && 
+            return searchInstCtrl.selectedInst &&
```
Comment (JuliePessoa) [inline frontend/user/searchInstitution/searchInstitution.component.js]: use lodash get

**Pull request merged**